python/tchannel/tornado/connection.py

```diff
@@ -75,13 +76,18 @@ class TornadoConnection(object):
     CALL_REQ_TYPES = frozenset([Types.CALL_REQ, Types.CALL_REQ_CONTINUE])
     CALL_RES_TYPES = frozenset([Types.CALL_RES, Types.CALL_RES_CONTINUE])
 
-    def __init__(self, connection):
+    def __init__(self, connection, tchannel=None):
         assert connection, "connection is required"
 
         self.closed = False
         self.connection = connection
+        try:
+            (self.remote_host,
+             self.remote_host_port) = connection.socket.getsockname()
+        except:
+            (self.remote_host, self.remote_host_port) = ("0.0.0.0", 0)
```

Comment:
1. Don't do catch-all like that.
2. If getsockname actually ever throws an exception, that's an actual problem and should be addressed. Not silently suppressed.